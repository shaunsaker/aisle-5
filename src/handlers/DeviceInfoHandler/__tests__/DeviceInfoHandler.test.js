import React from 'react';
import renderer from 'react-test-renderer';

import { DeviceInfoHandler } from '..';

describe('DeviceInfoHandler', () => {
  const spies = [];
  const dispatch = jest.fn();

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<DeviceInfoHandler dispatch={dispatch} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle getUniqueID', () => {
      const component = renderer.create(<DeviceInfoHandler dispatch={dispatch} />);
      const instance = component.getInstance();

      instance.getUniqueID();

      expect(dispatch).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();
    });
  });

  describe('lifecycle methods', () => {
    it('should call getUniqueID in componentDidMount', () => {
      spies[0] = jest.spyOn(DeviceInfoHandler.prototype, 'getUniqueID');
      renderer.create(<DeviceInfoHandler dispatch={dispatch} />);

      expect(spies[0]).toHaveBeenCalled();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
