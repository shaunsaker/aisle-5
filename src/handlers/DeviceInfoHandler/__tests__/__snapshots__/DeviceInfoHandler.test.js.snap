// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceInfoHandler methods should handle getUniqueID 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_UNIQUE_ID",
          },
        },
        "type": "getUniqueID",
      },
    ],
    Array [
      Object {
        "meta": Object {
          "nextAction": Object {
            "type": "SET_UNIQUE_ID",
          },
        },
        "type": "getUniqueID",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`DeviceInfoHandler renders renders with minimum required props 1`] = `null`;
